tests/yast2_gui/yast2_software_management.pm

```diff
@@ -9,16 +9,129 @@
+    send_key 'ret';
+    # make sure that package menu got closed after return
+    # otherwise following needle can be matched at wrong place
+    assert_screen [qw(all-in-this-list_still-selected sw_configurations)];
+    if (match_has_tag 'all-in-this-list_still-selected') {
+        mouse_click 'to-close-menu';
+    }
+    wait_still_screen;
+
+    # Configurations - Repositories, Online update
+    send_key 'alt-g';
+    assert_and_click 'sw_repositories';
+    assert_screen 'configured_sw_repo', 90;
+    # check PGP keys
+    send_key 'alt-g';
+    assert_screen 'gpg-public-key-management';
+    # cancel here and go back to main menu
+    assert_and_click 'configured_sw_gpg_cancel';
+    assert_and_click 'configured_sw_repo_cancel';
+
+    # Dependencies
+    assert_and_click 'sw_dependencies';
+    assert_and_click 'check_now';
+    assert_and_click 'all-packages-dependencies-ok';
+
+    # Options - Cleanup when deleting packages
+    assert_and_click 'sw_options';
+    assert_and_click 'cleanup-when-deleting-packages';
+
+    # Extras - Show Products, Changes, History
+    assert_and_click 'sw_extras';
+    assert_and_click 'show-products';
+    assert_and_click 'Products-ok';
+    assert_and_click 'sw_extras';
+    assert_and_click 'show-changes';
+    assert_and_click 'changed-packages-ok';
+    assert_and_click 'sw_extras';
+    assert_and_click 'show-history';
+    assert_and_click 'History-close';
+
+    # Help - Overview, Symbol, keys
+    assert_and_click 'sw_help';
+    assert_and_click 'help_overview';
+    assert_and_click 'overview-ok';
+    assert_and_click 'sw_help';
+    assert_and_click 'help-symbols';
+    assert_and_click 'symbols-ok';
+    assert_and_click 'sw_help';
+    assert_and_click 'special-keys-overview';
+    assert_and_click 'special-keys-overview-ok';
+
+    # View - Pattern
+    assert_and_click 'sw_view';
+    assert_and_click 'view_patterns';
+    assert_screen 'pattern-list';
+
+    # select a package to install: apache2
+    assert_and_click 'sw_search';
+    assert_and_click 'search_input';
+    type_string 'apache2';
+    send_key 'alt-s';
+    assert_and_click 'select-and-install';
+    assert_screen 'package-selected';
+
+    # RPM Groups - check Package Groups
+    assert_and_click 'rpm-groups';
+    assert_screen 'groups-list';
+
+    # Installation Summary
+    assert_and_click 'installation-summary';
+    assert_screen 'summary_details';
+
+    # Cover test cases like automatic changes and unsupported packages
+    send_key "alt-a";
+    wait_still_screen;
+    #Done and exit
+    assert_screen [qw(automatic-changes installation-report)];
+    if (match_has_tag('automatic-changes')) {
+        send_key 'alt-o';
+        assert_screen [qw(unsupported-packages installation-report)];
+        if (match_has_tag('unsupported-packages')) {
+            send_key 'alt-o';
+            wait_still_screen 120;
+            assert_and_click 'sw_finish';
+            wait_still_screen 60;
```

Comment:
wait_still_screen is a bad idea in gui tests and any stilltime with value 60 or 120 s does not make any sense as at least the clock timer on the desktop is changing once every 60 seconds so this call iteslf can never succeed so it is no better than a sleep 60s and giving a false impression